### PR TITLE
Fix pkgdown macro

### LINKED ISSUE #141: ci_is_tag() prevents deployment
Build: https://travis-ci.org/ropenscilabs/tic/jobs/519758834#L1391

https://github.com/ropenscilabs/tic/blob/a9da3f9501501438601a1e710e26eaa63c1c5a72/R/steps-git.R#L208

@krlmlr Any idea why this happens?

### PR DESCRIPTION
**Pull request by pat-s**
Should address all issues mentioned in https://github.com/ropenscilabs/tic/pull/126#pullrequestreview-226369606. 

Fixes #141 .

### CODE REVIEW

**Comment by krlmlr** [comment]
Deploy preview for *wonderful-volhard-261e08* ready!

Built with commit 33d72c065cabfe53c8fd1101928fcaeccdddab79

https://deploy-preview-142--wonderful-volhard-261e08.netlify.com

File: R/dsl.R [hunk 1]
```diff
@@ -156,15 +156,12 @@ do_package_checks <- function(...,
 #' @inheritParams step_setup_push_deploy
 #' @inheritParams step_do_push_deploy
 #' @param build_only Build the pkgdown site but do not deploy it.
```
Comment (krlmlr) [inline R/dsl.R]: This now needs a comment that the deployment never happens if we can't push. Maybe we should move the `ci_can_push()` check into the corresponding steps, and link explicitly to these steps?

File: R/dsl.R [hunk 2]
```diff
@@ -178,7 +175,9 @@ do_pkgdown <- function(...,
   get_stage("install") %>%
     add_step(step_install_deps(repos = repos))
 
-  if (isTRUE(build_only) || !deploy) {
+  needs_deploy <- !isTRUE(build_only) && ci_can_push()
+
+  if (isTRUE(needs_deploy)) {
```
Comment (krlmlr) [inline R/dsl.R]: The `isTRUE()` looks superfluous, here and below.

Please double-check the logic.

File: R/dsl.R [hunk 3]
```diff
@@ -197,7 +196,7 @@ do_pkgdown <- function(...,
   get_stage("deploy") %>%
     add_step(step_build_pkgdown(...))
 
-  if (isTRUE(build_only) || !deploy) {
+  if (isTRUE(needs_deploy)) {
```
Comment (krlmlr) [inline R/dsl.R]: Please double-check the logic.

File: R/dsl.R (hunk 1 above)
Comment (pat-s) [reply R/dsl.R]: Good idea, makes sense.

**Comment by pat-s** [comment]
Reworked the logic. Will continue tomorrow, guess I'll need some sleep ;) :sleeping:

**Comment by krlmlr** [comment]
Thanks! This fixes some logical errors. More things to look at later:

- Move the checks related to `ci_can_push()` and `ci_is_tag()` from the macro to the steps
- Improve documentation of the `build_only` argument, link to the steps
- Don't add the `step_install_deps()` step twice, if a `do_package_check()` is present (how?)

**Pull request merged**